app/components-react/windows/SourceProperties.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useMemo, useState } from 'react';
+import { Services } from '../service-provider';
+import { ObsForm } from '../obs/ObsForm';
+import { TObsFormData } from '../../components/obs/inputs/ObsInput';
+import { useSelector } from '../store';
+import { ModalLayout } from '../shared/ModalLayout';
+import Display from '../shared/Display';
+import { assertIsDefined } from '../../util/properties-type-guards';
+
+export default function SourceProperties() {
+  const { WindowsService, SourcesService } = Services;
+
+  // get source
+  const source = useMemo(() => {
+    const { sourceId } = WindowsService.getChildWindowQueryParams();
+    return SourcesService.views.getSource(sourceId);
+  }, []);
+
+  // define reactive variables
+  const [properties, setProperties] = useState(source ? source.getPropertiesFormData() : []);
```

Comment:
as i understand this will be running `getPropertiesFormData` every render, which we might want to avoid as i think it's a somewhat expensive operation

Reply:
Good callout.  This will happen on every render, and it is definitely an expensive operation.  The solution here is to pass a function as the argument to `useState` which returns the initial state.